lib/rspec/core/profiler.rb

```diff
@@ -0,0 +1,29 @@
+module RSpec
+  module Core
+    # @private
+    class Profiler
+      NOTIFICATIONS = [:example_group_started, :example_group_finished, :example_started]
+
+      def initialize
+        @example_groups = Hash.new { |h, k| h[k] = Hash.new(0) }
+      end
+
+      attr_reader :example_groups
+
+      def example_group_started(notification)
+        @example_groups[notification.group.id][:start] = Time.now
+        @example_groups[notification.group.id][:description] = notification.group.top_level_description
```

Comment:
Do you think there's an advantage to using the `id` as the hash key instead of the group itself?  The group is a class, which can be a hash key just fine, so the `id` seems like an unnecessary layer of indirection.  Plus there's some computation that has to be done to generate it so we can avoid that by using the `group` as the key.